katdal/datasources.py

```diff
@@ -97,15 +113,30 @@ class ChunkStoreVisFlagsWeights(VisFlagsWeights):
     """
     def __init__(self, store, base_name, chunk_info):
         self.store = store
-        da = {}
+        darray = {}
+        extra_flags = []
         for array, info in chunk_info.iteritems():
             array_name = store.join(base_name, array)
-            da[array] = store.get_dask_array(array_name, info['chunks'],
-                                             info['dtype'])
-        vis = da['correlator_data']
-        flags = da['flags']
+            chunk_args = (array_name, info['chunks'], info['dtype'])
+            darray[array] = store.get_dask_array(*chunk_args)
+            # Find all missing chunks in array and convert to 'data_lost' flags
+            has_array = store.has_dask_array(*chunk_args)
+            chunks_lost = da.map_blocks(_has_chunk_to_flags, has_array,
+                                        token='missing-chunks-' + array_name,
+                                        chunks=info['chunks'], dtype=np.uint8,
+                                        full_chunks=info['chunks'])
+            # If a flag chunk is missing but not the corresponding vis/weights,
+            # the data_lost bit will *not* be set (all flags cleared, actually)
+            if array != 'flags':
+                extra_flags.append(chunks_lost)
+                extra_flags.append('ijk'[:chunks_lost.ndim])
+        vis = darray['correlator_data']
+        # Combine original L0 flags with extras (missing chunks per array)
+        flags = da.atop(_multi_or_3d, 'ijk', darray['flags'], 'ijk',
+                        *extra_flags, token=store.join(base_name, 'flags_raw'),
```

Comment:
For me the cooked flags are the bools in `katdal.DataSet` (after the selected flag bits are ORed together). The uint8 version is 'raw'.

My thinking on the token was that this was a data product like `correlator_data` as opposed to an intermediate step. I can see that it could confuse someone.

Reply:
flags_raw makes sense to me now. Carry on.